content/200-concepts/100-components/02-prisma-client/063-excluding-fields.mdx

```diff
@@ -0,0 +1,69 @@
+---
+title: 'Excluding Fields'
+metaTitle: 'Excluding Fields'
+metaDescription: 'This page explains how to exclude sensitive fields from the Prisma Client'
+---
+
+<TopBlock>
+
+By default the Prisma Client returns all the fields back from a model. You can use `select` to narrow the result set, but that can be unwieldy if you have a large model and you only want to exclude one or two fields.
+
+The Prisma Client doesn't have a native way of excluding fields yet, but it's easy to create a function that you can use to exclude certain fields in a type-safe way.
+
+</TopBlock>
+
+## Excluding the Password Field
+
+The following is a type-safe `exclude` function returns a user without the `password` field.
+
+<TabbedContent tabs={[<FileWithIcon text="TypeScript" icon="code"/>, <FileWithIcon text="JavaScript (Node.js)" icon="code"/>]}>
+
+<tab>
+
+```tsx highlight=2-5;normal
+// Exclude keys from user
+function exclude<User, Key extends keyof User>(
+  user: User,
+  ...keys: Key[]
+): Omit<User, Key> {
+  for (let key of keys) {
+    delete user[key]
```

Comment:
Nit: Doesn't that risk of warning if the key doesn't exist in `user`?

Reply:
The `Key extends keyof User` bit removes the warning, telling Typescript that Key is a key within User. It's pretty magical.